src/DS/sds.c

```diff
@@ -848,17 +849,25 @@ static int ds_sds_compose_add_component_dependencies(xmlDocPtr doc, xmlNodePtr d
 
 				if (oscap_acquire_url_is_supported(href)) {
 					/* If the referenced component is remote one, do not include
-					 * it within the DataStream. Such component shall only be
-					 * downloaded once the scan is run. */
-					xmlFree(href);
-					continue;
+					* it within the DataStream. Such component shall only be
+					* downloaded once the scan is run.
+					* Add the component to the catalog and */
+					href_is_remote = true;
 				}
 
-				// skip over file:// if it's used in the file href
-				const char *altered_href = oscap_str_startswith(href, "file://") ? href + 7 : href;
+				char *real_path;
+				if (href_is_remote) {
+					// fix real_path for remote source
+					// Update href name to be only remote file (com.redhat.rhsa-RHEL7.xml)
+					// char *real_path = oscap_strdup(href)
+					real_path = oscap_strdup("com.redhat.rhsa-RHEL7.xml");
```

Comment:
I don't understand. If content references any remote content from anywhere, will it always get replaced by RHEL 7 RHSA? Hard coding anything here seems suspicious.

Reply:
I was just lazy to implement a URL parser before I could test that this is a good approach.